client/src/pages/queries/components/query-editor.tsx

```diff
@@ -176,7 +179,11 @@ export default function QueryEditor() {
                 </div>
               </Tabs.TabPane>
               <Tabs.TabPane tab="Charts" key="chart">
-                <div />
+                <EditorContext.Provider
+                  value={{ isSaved, queryId: id, queryResult }}
```

Comment:
Instead of creating `isSaved` variable as the top, we can set it over here `value={{isSaved: tabType === TabType.saved}}`